### PR TITLE
chore(flutter_form_validation): upgrade to Dart 2.17

### PR DESCRIPTION
**Pull request by hkuczynski**
## Status

**READY**

## Breaking Changes

NO

## Description

chore(flutter_form_validation): upgrade to Dart 2.17

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: examples/flutter_form_validation/analysis_options.yaml [hunk 1]
```diff
@@ -1,81 +1,6 @@
-analyzer:
-  strong-mode:
-    implicit-casts: false
-    implicit-dynamic: false
-  errors:
-    close_sinks: ignore
+include: package:very_good_analysis/analysis_options.3.0.0.yaml
```
Comment (wolfenrain) [inline examples/flutter_form_validation/analysis_options.yaml:1]:
```suggestion
include: package:very_good_analysis/analysis_options.3.0.1.yaml
```

File: examples/flutter_form_validation/lib/bloc/my_form_bloc.dart [hunk 2]
```diff
@@ -18,53 +19,67 @@ class MyFormBloc extends Bloc<MyFormEvent, MyFormState> {
 
   @override
   void onTransition(Transition<MyFormEvent, MyFormState> transition) {
+    // ignore: avoid_print
     print(transition);
     super.onTransition(transition);
   }
```
Comment (wolfenrain) [inline examples/flutter_form_validation/lib/bloc/my_form_bloc.dart:25]: Do we need to keep this in? 

/cc @felangel

File: examples/flutter_form_validation/lib/main.dart [hunk 3]
```diff
@@ -3,25 +3,30 @@ import 'package:flutter_bloc/flutter_bloc.dart';
...
     );
   }
 }
 
 class MyForm extends StatefulWidget {
+  const MyForm({super.key});
+
   @override
+  // ignore: library_private_types_in_public_api
   _MyFormState createState() => _MyFormState();
```
Comment (wolfenrain) [inline examples/flutter_form_validation/lib/main.dart:30]:
```suggestion
  State<MyForm> createState() => _MyFormState();
```

File: examples/flutter_form_validation/pubspec.yaml [hunk 4]
```diff
@@ -4,15 +4,16 @@ version: 1.0.0+1
...
 
 dependencies:
+  equatable: ^2.0.0
   flutter:
     sdk: flutter
   flutter_bloc:
     path: ../../packages/flutter_bloc
-  equatable: ^2.0.0
   formz: ^0.4.0
+  very_good_analysis: ^3.0.0
```
Comment (wolfenrain) [inline examples/flutter_form_validation/pubspec.yaml:16]:
```suggestion
  very_good_analysis: ^3.0.1
```

File: examples/flutter_form_validation/pubspec.yaml [hunk 5]
```diff
@@ -4,15 +4,16 @@ version: 1.0.0+1
 publish_to: none
 
 environment:
-  sdk: ">=2.16.0 <3.0.0"
+  sdk: ">=2.17.0 <3.0.0"
 
 dependencies:
+  equatable: ^2.0.0
```
Comment (wolfenrain) [inline examples/flutter_form_validation/pubspec.yaml:10]:
```suggestion
  equatable: ^2.0.3
```

**Review by wolfenrain — APPROVED** [review]

**Review by felangel — APPROVED** [review]

**Pull request merged**